### DISCUSSION TITLE
v158: paypalwpp, which language to use on PayPal redirect

### DISCUSSION THREAD

**Discussion by lat9** [zencart/zencart#5838] [Ideas]
Copied from #4985, so that it's not forgotten.

> > But still, the order of the loading of the array is effectively the language priority, so the browser language has priority over the session language....which I would disagree with.
> 
> While I agree that it appears 'funny' to be viewing a site in Spanish and then being redirected to PayPal in English, English _is_ my first language and the language of my browser and I'd find it a PITA to have to change the PayPal language back to English to complete my purchase on a Spanish-language site.
> 
> **Update**: Additionally, the store language will have precedence over the browser language, so the currently-loaded main language file (e.g. `spanish.php`) will have priority over the browser language, too. It feels to me that the browser-language should be the first entries in the to-be-checked array.
> 
> Other opinions welcomed!

To reiterate, it seems to me that the browser language should have precedence in any PayPal redirection, so that the PayPal payment display is in the customer's language, but I'm open to discussion.  @torvista?

**Comment by torvista** [comment]
I think the browser language should determine the shop language on the first page load only. 
I jump across half a dozen pcs where the operating system and browser language is dependent on who normally uses that computer, but on a website where **_I_** choose the language I think that should take preference throughout the navigation on that website.
Assuming the customers preferred language is that **chosen** in the shop, I would expect any embedded (incontext) content or auto-linked content to use that same **chosen** shop language to ensure the navigation/experience is "joined-up".

**Comment by torvista** [comment]
After spending some time with this
https://github.com/zencart/zencart/pull/5060
I concur that if it is to work "properly" there is a lot of time-consuming reviewing necessary regarding the hard-coded Paypal-accepted locales and countries and why these arrays are merged or wiped, when incontext is used or not and why, that happens when a country or locale is passed to Paypal and why and more... 
I've just hacked it for two languages based on the session.
Since no-one has ever complained, I suggest leaving it alone until they do, there are more pressing matters.

**Comment by scottcwilson** [comment]
Should this be closed?
